Description: Set opendkim-genkey to default to 2048 bit sha-256 only keys based
 on current best practices
Author: Scott Kitterman <scott@kitterman.com>
Forwarded: not-needed

--- a/opendkim/opendkim-genkey.8.in
+++ b/opendkim/opendkim-genkey.8.in
@@ -31,8 +31,9 @@
 (\-\-bits=n)
 Specifies the size of the key, in
 .I bits,
-to be generated.  The default is 1024 which is the value recommended by
-the DKIM specification.
+to be generated.  The upstream default is 1024 which is the value recommended
+by the DKIM specification, but in Debian the default is 2048 based on more
+current recommendations such as those from NIST 800-177.
 
 .TP
 .I \-d domain
@@ -54,7 +55,8 @@
 (\-\-hash-algorithms=name[:name[...]])
 Specifies a list of hash
 .I algorithms
-which can be used with this key.  By default all hash algorithms are allowed.
+which can be used with this key.  Upstream, by default all hash algorithms are
+allowed, but in Debian this is restricted to sha256 based on NIST 800-177.
 
 .TP
 .I \-\-help
--- a/opendkim/opendkim-genkey.in
+++ b/opendkim/opendkim-genkey.in
@@ -28,13 +28,15 @@
 ## Set up defaults
 ##
 
-my $bits = 1024;
+# Debian patch to set default to 2048 based on 2016 NIST recommendations:
+my $bits = 2048;
 my $domain = "@DOMAIN@";
 my $version = "@DOMAIN@";
 my $outdir = ".";
 my $nosubdomains = 0;
 my $subdomains = 1;
-my $hashalgs;
+# Debian patch to use sha256 only based on 2016 NIST recommendations:
+my $hashalgs = "sha256";
 my $note;
 my $flags = "";
 my $selector = "default";
